xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/internal/authentication/LoginLinkUIExtension.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.internal.authentication;
```

Comment:
AFAIK the practice is to have **a single** internal package per module. I know the [naming rule says](https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HPackagenames) to put the module name before ``internal`` but in this case the module is ``oldcore`` and since this is the first internal package I saw in ``oldcore`` (inside ``org.xwiki``) I assumed this is the internal package of ``oldcore``. Since you asked this question I checked, and it seems there are other internal packages in ``oldcore`` (``evaluation.internal``, ``model.internal``, ``security.internal``) which seems wrong to me.

So if we were to follow the rule the package name should have been ``org.xwiki.oldcore.internal.authentication``, but I'm not sure if it's worth introducing yet another internal package, unless we rename the existing ``org.xwiki.internal`` into ``org.xwiki.oldcore.internal`` but I don't think it's a good idea to perform this rename in this PR for 16.10.0.

Reply:
Ok I haven't checked the structure in oldcore tbh. And I was thinking about packages in security module where it's probably `org.xwiki.security.authentication` anyway, so let's keep it like that.